packages/typespec-client-generator-core/src/decorators.ts

```diff
@@ -113,20 +124,53 @@ function setScopedDecoratorData(
     return;
   }
 
-  // if scope specified, create or overwrite with the new value
-  const splitScopes = scope.split(",").map((s) => s.trim());
-  const targetEntry = context.program.stateMap(key).get(target);
-
-  // if target doesn't exist in decorator map, create a new entry
-  if (!targetEntry) {
-    const newObject = Object.fromEntries(splitScopes.map((scope) => [scope, value]));
+  const negationScopes = getNegationScopes(scope);
+  if (negationScopes !== undefined && negationScopes.length > 0) {
+    const newObject = Object.fromEntries([AllScopes].map((scope) => [scope, value]));
+    newObject[negationScopesKey] = negationScopes;
     context.program.stateMap(key).set(target, newObject);
     return;
+  } else {
+    const splitScopes = scope.split(",").map((s) => s.trim());
+
+    // if negation scope is combined with normal scope, report error
+    if (splitScopes.length > 1 && splitScopes.some((s) => s.startsWith("!"))) {
+      reportDiagnostic(context.program, {
+        code: "invalid-negation-scope",
+        target: context.decoratorTarget,
+      });
+      return;
+    }
+
+    // if scope specified, create or overwrite with the new value
+    const targetEntry = context.program.stateMap(key).get(target);
+
+    // if target doesn't exist in decorator map, create a new entry
+    if (!targetEntry) {
+      const newObject = Object.fromEntries(splitScopes.map((scope) => [scope, value]));
+      context.program.stateMap(key).set(target, newObject);
+      return;
+    }
+
+    // if target exists, overwrite existed value
+    const newObject = Object.fromEntries(splitScopes.map((scope) => [scope, value]));
+    context.program.stateMap(key).set(target, { ...targetEntry, ...newObject });
   }
+}
 
-  // if target exists, overwrite existed value
-  const newObject = Object.fromEntries(splitScopes.map((scope) => [scope, value]));
-  context.program.stateMap(key).set(target, { ...targetEntry, ...newObject });
+function getNegationScopes(scope?: LanguageScopes): string[] | undefined {
+  if (scope === undefined) {
+    return undefined;
+  }
+
+  const negationScopeRegex = new RegExp(/!\((.*?)\)/);
+  const negationScopeMatch = scope.match(negationScopeRegex);
+  if (negationScopeMatch) {
+    return negationScopeMatch[1].split(",").map((s) => s.trim());
+  } else if (!scope.includes(",") && scope.startsWith("!")) {
```

Comment:
since we have two kinds of patten for scope, can we have a regex that could cover all cases and make sure all user's input are valid, else report diagnostic?

Reply:
Based on https://github.com/Azure/typespec-azure/pull/1783#discussion_r1825460771, we do have 2 syntax formats.
```
!(scope1, scope2, ...)
```
and
```
!scope1, !scope2, ...
```
I think it's clearer to handle them separately. Please take another look at the updated implementation.